options/components/webTopContent/top-contents/WebKeywordSearchBox.jsx

```diff
@@ -0,0 +1,130 @@
+import { faMagnifyingGlass, faRotate } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useContext, useState } from "react";
+
+import { WebSearchContext } from "../../../context/WebSearchContext";
+
+export default function WebKeywordSearchBox({ selectData, urlNewList }) {
+  const [currentSelectValue, setCurrentSelectValue] = useState(
+    selectData[0].selectType
+  );
+
+  return (
+    <div className="flex">
+      <WebSelectBox
+        selectData={selectData}
+        setCurrentSelectValue={setCurrentSelectValue}
+        currentSelectValue={currentSelectValue}
+      />
+      <WebSearchBox urlNewList={urlNewList} />
+    </div>
+  );
+}
+
+function WebSelectBox({
+  selectData,
+  setCurrentSelectValue,
+  currentSelectValue,
+}) {
+  const [isShowOptions, setIsShowOptions] = useState(false);
+
+  const handleOnChangeSelectToggle = () => {
+    return setIsShowOptions(!isShowOptions);
+  };
+
+  const handleOnChangeSelectValue = (e) => {
+    const { innerText } = e.target;
+    setCurrentSelectValue(innerText);
+  };
+
+  return (
+    <div
+      className={
+        "relative w-[200px] h-10 px-[8px] me-3 rounded-lg bg-black cursor-pointer text-white after:content-['⌵'] after:absolute after:top-2 after:right-3"
+      }
+      onClick={() => handleOnChangeSelectToggle()}
+    >
+      <label className=" text-sm leading-10">{currentSelectValue}</label>
+      <div
+        className={
+          isShowOptions === false
+            ? "absolute overflow-hidden list-none left-0 w-full max-h-[0px]"
+            : "absolute overflow-hidden list-none rounded-b-lg top-[35px] left-0 w-full h-[108px] max-h-none bg-black text-white text-sm"
+        }
```

Comment:
테일윈드로 스타일에 대해 조건을 걸어주신 점 좋게 생각합니다, 다만 option이 false일때와 true일 때에 대해 css가 바뀌는 부분인 것 같은데 positiion, overflow-hidden등 공통되는 속성에 대해서도 같이 조건을 걸어줘야 하는지 의문입니다. 같은 스타일이 중첩되어 있어 가독성을 조금 저하시키는 것 같습니다.

Reply:
말씀 주신 부분 참조하여 아래와 같이 css 중첩을 줄이고 정리하였습니다.
```suggestion
        className={
          isShowOptions === false
            ? "hidden"
            : "block absolute overflow-hidden list-none rounded-b-lg top-[35px] left-0 w-full h-[108px] max-h-none bg-black text-white text-sm"
        }
```